MMR.Randomizer/Utils/SequenceUtils.cs

```diff
@@ -777,6 +782,66 @@ private static void RelocateSeq(int f)
             ReadWriteUtils.WriteToROM(0x00C2739C, new byte[] { 0x3C, 0x08, 0x80, 0x0A, 0x8D, 0x05, (byte)(offset >> 8), (byte)(offset & 0xFF) });
         }
 
+        public static void MoveAudioBankTableToFile()
+        {
+            // grab original audiobanktable out of code, plus extra for modifying
+            var table = ReadWriteUtils.ReadBytes(0xB3C000 + 0x13B6C0, 0x820);
+            New_AudioBankTable = RomUtils.AddNewFile(table);
+
+            // instrumentset_patch: modifies audiobank metadata read and writes, instrument/drum/sfx pointer read and writes,
+            // nops a metadata copy function, and sets a fixed size for the audiobank pointer index
+            ResourceUtils.ApplyHack(Resources.mods.instrumentset_patch);
+
+            // moveaudiostatebytes: sets where read and writes for sequence and instrumentset states go
+            // in this hack, they're moved from 0x80205008 to end of old instrumentset table in code and given more space
+            // if these don't get moved, new banks at 0x30 and up will overflow into sequence states and can knock out sound
+            ResourceUtils.ApplyHack(Resources.mods.moveaudiostatebytes);
+
+            // loadnewaudiotable: where the copy metadata function loop was, sets a jump to code placed at the old instrumentset list
+            // which DMAs new audiobanktable from a file, relocates the addresses in the table, and sets the
+            // instrumentset table pointer to the new file
+            ResourceUtils.ApplyHack(Resources.mods.loadnewaudiotable);
+
+            // can't update addresses in an audiobank table that's moved and not loaded yet
+            ReadWriteUtils.WriteCodeNOP(0x80190E70);
+            ReadWriteUtils.WriteCodeNOP(0x80190E74);
+            ReadWriteUtils.WriteCodeNOP(0x80190E78);
+            ReadWriteUtils.WriteCodeNOP(0x80190E7C);
+            ReadWriteUtils.WriteCodeNOP(0x80190E80);
+
+            // To do: make this dynamic with payload length, at some point. maybe.
+            ReadWriteUtils.WriteU32ToROM(0xC776C0, 0x807343D0); //RAM address to move audiobanktable into
```

Comment:
Where does the value `0x807343D0` come from?

Reply:
It's where the asm payload is loaded into ram plus its size (0x80720000 + 143D0) (or was it 143B0?). I didn't know how to check for it so it's hardcoded like that.